In [6]:
import pandas as pd

#Data set 1: Musica y salud mental
from google.colab import drive
import os
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [20]:
%cd '/content/gdrive/My Drive'
df= pd.read_excel('music_mentalhealth.xlsx')
print(df.head(5))


/content/gdrive/My Drive
                        Age Primary streaming service  Hours per day  \
0  8/27/2022 19:29:02  18.0                   Spotify            3.0   
1  8/27/2022 19:57:31  63.0                   Pandora            1.5   
2  8/27/2022 21:28:18  18.0                   Spotify            4.0   
3  8/27/2022 21:40:40  61.0             YouTube Music            2.5   
4  8/27/2022 21:54:47  18.0                   Spotify            4.0   

  While working Instrumentalist Composer         Fav genre Exploratory  \
0           Yes             Yes      Yes             Latin         Yes   
1           Yes              No       No              Rock         Yes   
2            No              No       No  Video game music          No   
3           Yes              No      Yes              Jazz         Yes   
4           Yes              No       No               R&B         Yes   

  Foreign languages  ...  Frequency [R&B]  Frequency [Rap] Frequency [Rock]  \
0               Ye

In [21]:
df.head()

Age Primary streaming service  Hours per day  \
0  8/27/2022 19:29:02  18.0                   Spotify            3.0   
1  8/27/2022 19:57:31  63.0                   Pandora            1.5   
2  8/27/2022 21:28:18  18.0                   Spotify            4.0   
3  8/27/2022 21:40:40  61.0             YouTube Music            2.5   
4  8/27/2022 21:54:47  18.0                   Spotify            4.0   

  While working Instrumentalist Composer         Fav genre Exploratory  \
0           Yes             Yes      Yes             Latin         Yes   
1           Yes              No       No              Rock         Yes   
2            No              No       No  Video game music          No   
3           Yes              No      Yes              Jazz         Yes   
4           Yes              No       No               R&B         Yes   

  Foreign languages  ...  Frequency [R&B]  Frequency [Rap] Frequency [Rock]  \
0               Yes  ...        Sometimes  Very frequently            Never   
1                No  ...        Sometimes           Rarely  Very frequently   
2               Yes  ...            Never           Rarely           Rarely   
3               Yes  ...        Sometimes            Never            Never   
4                No  ...  Very frequently  Very frequently            Never   

  Frequency [Video game music] Anxiety Depression Insomnia  OCD Music effects  \
0                    Sometimes     3.0        0.0      1.0  0.0           NaN   
1                       Rarely     7.0        2.0      2.0  1.0           NaN   
2              Very frequently     7.0        7.0     10.0  2.0     No effect   
3                        Never     9.0        7.0      3.0  3.0       Improve   
4                       Rarely     7.0        2.0      5.0  9.0       Improve   

     Permissions  
0  I understand.  
1  I understand.  
2  I understand.  
3  I understand.  
4  I understand.  

[5 rows x 33 columns]

Hours per day: Esta columna indica la cantidad de horas destinadas a escuchar musica por dia. Tiene potencial ya que cuanto mayor sean las horas, mejor es la salud mental del individuo. 


In [33]:
#Data set 2: Sleep quality
import pandas as pd

from google.colab import drive
import os
drive.mount('/content/gdrive')
%cd '/content/gdrive/My Drive'
df= pd.read_excel('Sleepdataex.xlsx')
print(df.head(5))


Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).
/content/gdrive/My Drive
                Start                 End  Sleep quality Time in bed Wake up  \
0                 NaT                 NaT            NaN         NaN     NaN   
1 2014-12-29 22:57:49 2014-12-30 07:30:13           1.00    08:32:00      :)   
2                 NaT                 NaT            NaN         NaN     NaN   
3 2014-12-30 21:17:50 2014-12-30 21:33:54           0.03    00:16:00      :|   
4                 NaT                 NaT            NaN         NaN     NaN   

     Sleep Notes  Heart rate  Activity (steps)  
0            NaN         NaN               NaN  
1            NaN        59.0               0.0  
2            NaN         NaN               NaN  
3  Stressful day        72.0               0.0  
4            NaN         NaN               NaN  


In [39]:
df.dropna(subset = ['Start']) 

Start                 End  Sleep quality Time in bed  \
1    2014-12-29 22:57:49 2014-12-30 07:30:13           1.00    08:32:00   
3    2014-12-30 21:17:50 2014-12-30 21:33:54           0.03    00:16:00   
5    2014-12-30 22:42:49 2014-12-31 07:13:31           0.98    08:30:00   
7    2014-12-31 22:31:01 2015-01-01 06:03:01           0.65    07:32:00   
9    2015-01-01 22:12:10 2015-01-02 04:56:35           0.72    06:44:00   
...                  ...                 ...            ...         ...   
1765 2018-02-12 21:54:14 2018-02-13 07:02:15           0.91    09:08:00   
1767 2018-02-13 23:49:19 2018-02-14 07:00:53           0.81    07:11:00   
1769 2018-02-14 21:24:05 2018-02-15 06:20:52           0.71    08:56:00   
1771 2018-02-15 21:36:32 2018-02-16 06:50:31           0.80    09:13:00   
1773 2018-02-16 22:52:29 2018-02-17 07:48:04           0.91    08:55:00   

     Wake up             Sleep Notes  Heart rate  Activity (steps)  
1         :)                     NaN        59.0               0.0  
3         :|           Stressful day        72.0               0.0  
5         :|                     NaN        57.0               0.0  
7        NaN                     NaN         NaN               0.0  
9         :)  Drank coffee:Drank tea        68.0               0.0  
...      ...                     ...         ...               ...  
1765     NaN                     NaN         NaN              56.0  
1767     NaN                     NaN         NaN              64.0  
1769     NaN                     NaN         NaN            3316.0  
1771     NaN                     NaN         NaN            6555.0  
1773     NaN                     NaN         NaN            2291.0  

[887 rows x 8 columns]

Time in bed: Esta columna indica la cantidad de horas dormidas. Tiene potencial ya que cuanto mayor sean las horas, mejor es la calidad de sueño del individuo.

In [6]:
#Data set 3: Nutricion

import pandas as pd

from google.colab import drive
import os
drive.mount('/content/gdrive')
%cd '/content/gdrive/My Drive'
df= pd.read_csv('Nutrition.csv',sep=',')
print(df.head(5))

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).
/content/gdrive/My Drive
     Company   Category                           Product Per Serve Size  \
0  Pizza Hut  All Meals          Corn n Cheese (Personal)        143.5 g   
1  Pizza Hut  All Meals          Country Feast (Personal)          178 g   
2  Pizza Hut  All Meals          Double Cheese (Personal)          143 g   
3  Pizza Hut  All Meals  Double Paneer Supreme (Personal)        174.5 g   
4  Pizza Hut  All Meals          Farmer`s Pick (Personal)          177 g   

   Energy (kCal)  Carbohydrates (g)  Protein (g)  Fiber (g)  Sugar (g)  \
0         432.60              65.64        17.91       3.85        0.0   
1         407.60              67.11        16.73       7.19        0.0   
2         423.33              59.97        18.26       3.49        0.0   
3         474.03              52.86        20.07       3.79        0.0   
4         408.16 

In [9]:
df.head()

Company   Category                           Product Per Serve Size  \
0  Pizza Hut  All Meals          Corn n Cheese (Personal)        143.5 g   
1  Pizza Hut  All Meals          Country Feast (Personal)          178 g   
2  Pizza Hut  All Meals          Double Cheese (Personal)          143 g   
3  Pizza Hut  All Meals  Double Paneer Supreme (Personal)        174.5 g   
4  Pizza Hut  All Meals          Farmer`s Pick (Personal)          177 g   

   Energy (kCal)  Carbohydrates (g)  Protein (g)  Fiber (g)  Sugar (g)  \
0         432.60              65.64        17.91       3.85        0.0   
1         407.60              67.11        16.73       7.19        0.0   
2         423.33              59.97        18.26       3.49        0.0   
3         474.03              52.86        20.07       3.79        0.0   
4         408.16              53.93        19.91       2.46        0.0   

   Total Fat (g)  Saturated Fat (g)  Trans Fat (g)  Cholesterol (mg)  \
0          10.93               5.14           0.16             16.19   
1           8.03               3.24           0.11             66.80   
2          12.27               5.23           0.18             19.75   
3          20.26               9.25           0.33             71.72   
4          12.53               4.90           0.14             48.09   

   Sodium (mg)  
0       499.72  
1       818.00  
2       638.22  
3      1128.11  
4       942.67

Saturated Fat: Esta columna indica la cantidad de gramos de grasa saturada que contiene la porcion servida. Tiene potencial para demostrar que la comida rapida no es sana ya que idealmente el peso de la grasa saturada no debe exceder del 10 porciento de las calorias. Sabiendo que cada gramo de grasa saturada equivale a 9 calorias, si tomasemos como ejemplo la primera fila, el 10% de 432.60 calorias es 43.26 calorias, y la equivalencia de esta en gramos de grasa saturada es 4.80 gr. 